### PR TITLE
Dashboard: Prevent folder from changing when navigating to general settings

### LINKED ISSUE #37348: "General" listed as destination folder sometimes while saving dashboards if the environment has 2000+ folders
**What happened**:
We have large Grafana installation with nearly 12,000 dashboards, since our Grafana v8 upgraded, "General" folder became invisible. This caused a major issue to us: since we have a dozen users are admin, so they have edit permission to "General" folder, a pre-existing issue that when admin review user dashboard and save the dashboard, sometimes it will be saved under "General" even without changing dashboard folder destination(this probably warrant another issue if not existed already) and often admins wouldn't even realize. When this happens, since we can't see General folder anymore, we don't even know what's in the General folder.

**What you expected to happen**:
1. General Folder should still be viewable/editable by admin
2. Saving an existing dashboard from regular folder by admin should not cause the dashboard be saved under General folder.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Grafana version: v8.0.6
- Data source type & version: Metrictank
- OS Grafana is installed on:
- User OS & Browser: RHEL 7.6 / Chrome
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
On very large Grafana installations with over 1000 folders and dashboards, the `FolderPicker` in `Dashboard => General Settings` would, under certain conditions (read details in #37348), revert to `General` because the matching folder wouldn't be found in the options returned by the backend api. This might be caused the limit of 1000 in the api and if the folder is not found in that result `FolderPicker` will revert to the `General` folder.

This PR addresses this by introducing a new property called [`skipInitialLoad`](https://github.com/grafana/grafana/blob/02d5bc84ca72a7dddd852d821c40d9119337fa9d/public/app/core/components/Select/FolderPicker.tsx#L28) to the `FolderPicker`.  When `skipInitialLoad` is used we will bypass the load of all folders and use the provided `initialFolderId` and `initialTitle` instead. 

- If `skipInitialLoad` is used and [`initialTitle`](https://github.com/grafana/grafana/blob/02d5bc84ca72a7dddd852d821c40d9119337fa9d/public/app/core/components/Select/FolderPicker.tsx#L17) is missing, then an api call is made to `/api/folders/id/initialFolderId`. 
- If `skipInitialLoad` is used and [`initialFolderId`](https://github.com/grafana/grafana/blob/02d5bc84ca72a7dddd852d821c40d9119337fa9d/public/app/core/components/Select/FolderPicker.tsx#L18) is missing, then an error is thrown. 

The `FolderPicker` component is got a bit out of control and I would rather have refactored out several (maybe) different FolderPicker components from it. But as this is supposed to be a bug fix, I've created this issue instead #38102 to handle that work. 

**Which issue(s) this PR fixes**:
Fixes #37348

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/manage-dashboards/state/actions.ts [hunk 1]
```diff
@@ -222,6 +223,14 @@ export function createFolder(payload: any) {
   return getBackendSrv().post('/api/folders', payload);
 }
 
+export function searchFolders(query: any, permission?: 'View' | 'Edit'): Promise<DashboardSearchHit[]> {
```
Comment (dprokop) [inline public/app/features/manage-dashboards/state/actions.ts:226]: Nit: `'View' | 'Edit'` - can we make it an enum?

**Review by dprokop — APPROVED** [review]
Looks and works good!

File: public/app/features/manage-dashboards/state/actions.ts (hunk 1 above)
Comment (hugohaggmark) [reply public/app/features/manage-dashboards/state/actions.ts:226]: Good suggestion, I'll fix it!

**Review by dsotirakis — APPROVED** [review]
Works as expected! 🚀

File: public/app/features/manage-dashboards/state/actions.ts (hunk 1 above)
Comment (hugohaggmark) [reply public/app/features/manage-dashboards/state/actions.ts]: Fixed in https://github.com/grafana/grafana/pull/38103/commits/a96a50ed929511dcdbfcd2e91e0cfdb027a4e4c2

**Pull request merged**